plugins/logitech-bulkcontroller/fu-plugin-logitech-bulkcontroller.c

```diff
@@ -0,0 +1,291 @@
+                        break;
+                case TRANSFER_FAILED:
+                        g_warning("[%s] : File transfer failed for interface %u", user_data->prog_name, bulk_intf);
+                        g_cond_signal(&user_data->test_upd_cond);
+                        break;
+                case TRANSFER_INPROGRESS:
+                        if (BULK_INTERFACE_UPD == bulk_intf) {
+                                g_print ("\r[%s] : Interface: %u File uploaded : [%d%%]", user_data->prog_name, bulk_intf, progress);
+                                fflush(stdout);
+                                FuDevice *device = (FuDevice *)user_data->device_ptr;
+                                if (device)
+                                  fu_device_set_progress (device, progress);
+                        } else if (BULK_INTERFACE_SYNC == bulk_intf) {
+                                g_print("\r[%s] : Interface: %u File received : [%d%%]", user_data->prog_name, bulk_intf, progress);
+                                fflush(stdout);
+                        }
+                        if (progress == 100) {
+                                g_print("\n");
+                        }
+                        break;
+                case TRANSFER_COMPLETED:
+                        g_debug("[%s] :  File transfer completed for interface %u", user_data->prog_name, bulk_intf);
+                        g_cond_signal(&user_data->test_upd_cond);
+                        break;
+                default:
+                        break;
+        }
+}
+
+static gchar *
+get_device_version (LogiBulkController *obj, void *data) 
+{
+        Message message;
+        gint ret;
+        ReturnValue *ret_val;
+        ApiUserData *user_data = (ApiUserData*) data;
+        g_cond_init (&user_data->test_upd_cond);
+        g_mutex_init (&user_data->test_upd_mutex);
+        g_mutex_lock (&user_data->test_upd_mutex);
+        
+        memset(&message,'\0',sizeof(message));
+        
+        ret = proto_manager_generate_get_device_info_request(&message);
+        if (!ret && message.data) {
+                LOGFN;
+                g_debug("Sending %s\n", message.data);
+                
+                ret_val = logibulkcontroller_send_data_sync(obj, message.data, message.len);
+                if ((ERRORCODE_NO_ERROR != ret_val->error_code) || (ERRORCODE_SEND_DATA_REQUEST_PUSHED_TO_QUEUE != ret_val->error_code)) {
+                        g_warning ("Error in send data %u", ret_val->error_code);
+                }
+                g_free(message.data);
+        }
+
+        if (g_cond_wait_until (&user_data->test_upd_cond, &user_data->test_upd_mutex,
+                                                        g_get_monotonic_time () + 5 * G_TIME_SPAN_SECOND)) {
+                g_mutex_unlock (&user_data->test_upd_mutex);
+                g_mutex_lock (&user_data->test_upd_mutex);
+        }
+        sleep(3); // For testing
+        return "1.2.3";
```

Comment:
you can't return a const string from something that's returning an allocated string